### ISSUE TITLE
[Audit_RV] Rounding error for the LRNA imbalance when buying with LRNA.

### ISSUE THREAD

**Issue by virgil-serbanuta** [galacticcouncil/HydraDX-math#68]
The rounding error for the LRNA imbalance when buying with LRNA can be fairly large, increasing the value of LRNA (over time, as the imbalance gets paid) and reducing profits from HDX fees.

The lower bound for the LRNA imbalance rounding error can be approximated by ([x] is the unrounded value of x)
`-Delta Balance_LRNA / ([hub_denominator] - 1) - something`. The "something" part is usually not that large. However, in principle, the `[hub_denominator]` value can be rather small, e.g. it can be `2`, which means that the total rounding error can be large.

**Comment by virgil-serbanuta** [comment]
I'm still rechecking my computation, but the formula above applies only to the initial audit, not to the audit for the version which include the first batch of fixes.

There are two errors that matter: the error for delta_hub_reserve, and the error for delta_imbalance.

If the hub_denominator value is rounded down (the version with fixes does this), then it seems that
```
error(delta_hub_reserve) is in [0, asset_out_amount / ([hub_denominator] - 1) + 1)
error(delta_imbalance) is in [0, max(error(delta_hub_reserve))*(1+complicated-stuff))
```
This means that delta_hub_reserve is rounded in favor of the pool (the contract receives more than the unrounded value) and delta_imbalance is also rounded in favor of the pool (the imbalance is larger than the unrounded value, which means that, in the end, the LRNA value will grow).

If this turns out to be correct, then this issue should be closed.

**Comment by enthusiastmartin** [comment]
fixed in #70
